tests/integration/security/util/framework.go

```diff
@@ -48,4 +48,10 @@ func EchoConfig(name string, ns namespace.Instance, headless bool, annos echo.An
 		Galley: g,
 		Pilot:  p,
 	}
+
+	// for headless service, the port and target port must be equal
```

Comment:
if the comment below is correct, *they must be equal* is just to accommodate one situation, which worth additional explaining. *must be equal*, because under mTLS,...., fails; under plain text, .... works.

ideally e2e test should be expressive enough to configure both. this seems more of actual test code's logic, i have mtls, i want to go through all cases a b c, rather than hardcoded into framework.

Reply:
@incfly No, this comment means the headless service port and targetPort must be equal, which is also a requirement of pure k8s. Otherwise, imagine when you connect to a pod with podip + servicePort, definitely you will get connection refused.

Reply:
comment doc, it'll be great to link k8s headless doc link here.